Automattic/wp-calypso
# Themes: Remove more obsolete theme actions, take two

Remove theme actions that only consist of a `dispatch` (that doesn't trigger any reducers), and of a `navigateTo` or `page` call, i.e. navigating to a given URL. Instead, pass those URLs as `href`s to relevant `a`s.

Preparatory work for #6165. Addresses item # 3 (Avoid side-effects in action creators whenever possible) of #4057.

Supersedes #6339.

To test:
* Check that every possible way in the showcase to trigger signup or customization really works. This includes:
 * 'More' button items
 * Site selector modal primary button
 * Preview primary button
* Verify that the other options still work, too.
* Pay special attention to the Thanks modal appearing when activating a theme via 'Try & Customize' > 'Save & Activate'
* Actually, probably best to go through the [entire list
](https://github.com/Automattic/wp-calypso/pull/6349#issuecomment-232339021) since this PR potentially touches all options. It's the last one to do so for a while, I promise.

Aside: I've noticed that Jetpack sites have a 'Live Demo' option that does the same as 'Try & Customize'. @folletto do we want to get rid of either (just for JP sites, that is)? (I think the logic would also be quite a bit cleaner without one of them, probably 'Live Demo'.)

Test live: https://calypso.live/?branch=remove/more-obsolete-theme-actions-take-two
